### PR TITLE
Implemented PredefinedType Consistency issue 129

### PR DESCRIPTION
**Pull request by SergejMuhic**
1. Subtypes of `IfcGeotechnicalStratum` were deleted and replaced with a `PredefinedType` : `IfcGeotechnicalStratumTypeEnum`
2. Applicabilities of IfcGeotechnicalStratum psets were amended
3. `IfcPlant` was removed in favour of `IfcGeographicElement` having a `PredefinedType` attribute
4. Status of `IfcGeographicElementTypeEnum` changed back to active from "deprecated"
5. `Pset_PlantCommon` renamed to `Pset_VegetationCommon`
6. All touched psets received `PropertySetType` = `PSET_OCCURRENCEDRIVEN`

### CODE REVIEW

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechnicalStratum/DocEntity.xml [hunk 1]
```diff
@@ -1,9 +1,20 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeotechnicalStratum" UniqueId="faefb134-3800-4995-b222-b921d7e287bf" BaseDefinition="IfcGeotechnicalElement">
-	<Subtypes>
-		<DocSubtype UniqueId="964a446c-3ce6-44e6-b70b-cc43a1ff39bd" DefinedType="IfcSolidStratum" />
-		<DocSubtype UniqueId="ec4f4f65-4341-4538-8fd3-c067f3bd6986" DefinedType="IfcVoidStratum" />
-		<DocSubtype UniqueId="fb21c163-8d7a-45cf-b517-421f586727b0" DefinedType="IfcWaterStratum" />
-	</Subtypes>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeotechnicalStratum" UniqueId="faefb134-3800-4995-b222-b921d7e287bf" BaseDefinition="IfcGeotechnicalElement" EntityFlags="32">
```
Comment (pjanck) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechnicalStratum/DocEntity.xml:2]: What is EntityFlags?

**Review by pjanck — APPROVED** [review]
Lftm

File: IFC4x3/Constants/v/VEGETATION_039IZnw4P17vEJksQmuxAR.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VEGETATION_039IZnw4P17vEJksQmuxAR" Name="VEGETATION" UniqueId="032528f1-e846-411f-9393-bb66b0e3b29b">
+	<Documentation>Plant life or plant cover (as of an area). For example trees, shrubs, herbs, grasses, ferns, and mosses.</Documentation>
+</DocConstant>
+
```
Comment (larswik) [inline IFC4x3/Constants/v/VEGETATION_039IZnw4P17vEJksQmuxAR.xml:5]: I think that IfcPlant was primarily aimed towards something man-made and not existing vegetation. Is this distinction clear?

File: IFC4x3/Constants/s/SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4.xml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4" Name="SOIL_BORING_POINT" UniqueId="db1fd97e-02d2-48d3-ab67-898b2f960384" Status="Deprecated" />
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4" Name="SOIL_BORING_POINT" UniqueId="db1fd97e-02d2-48d3-ab67-898b2f960384" />
```
Comment (larswik) [inline IFC4x3/Constants/s/SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4.xml:3]: Distinction between this and an IfcBorehole?

File: IFC4x3/Constants/s/SOLID_0hCk8da6HBZAf0CKpzqv2y.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SOLID_0hCk8da6HBZAf0CKpzqv2y" Name="SOLID" UniqueId="2b32e227-9064-4b8c-aa40-314cfdd390bc">
+	<Documentation>Representation of the concept of an identified discrete almost homogenous solid geological or surface feature, including discontinuities such as faults, fractures, boundaries and interfaces that are not explicitly modelled.</Documentation>
+</DocConstant>
+
```
Comment (larswik) [inline IFC4x3/Constants/s/SOLID_0hCk8da6HBZAf0CKpzqv2y.xml:5]: IfcGeoScienceFeature from IfcTunnel will be covering this in another way.

File: IFC4x3/Constants/t/TERRAIN_2TKelKxjbD$POIgGNYWWXi.xml [hunk 5]
```diff
@@ -1,3 +1,3 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TERRAIN_2TKelKxjbDPOIgGNYWWXi" Name="TERRAIN" UniqueId="9d528bd4-eed9-4dfd-9612-a905e282086c" Status="Deprecated" />
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TERRAIN_2TKelKxjbDPOIgGNYWWXi" Name="TERRAIN" UniqueId="9d528bd4-eed9-4dfd-9612-a905e282086c" />
```
Comment (larswik) [inline IFC4x3/Constants/t/TERRAIN_2TKelKxjbD$POIgGNYWWXi.xml:3]: IfcGeoScienceModel from IfcTunnel will be covering this in another way.

File: IFC4x3/Constants/v/VOID_2hj_ExWTf21wnmjHgzcx24.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="VOID_2hj_ExWTf21wnmjHgzcx24" Name="VOID" UniqueId="abb7e3bb-81da-4207-ac70-b51abd9bb084">
+	<Documentation>Representation of the concept of an identified discrete air filled geological feature, including caves and other voids.</Documentation>
+</DocConstant>
+
```
Comment (larswik) [inline IFC4x3/Constants/v/VOID_2hj_ExWTf21wnmjHgzcx24.xml:5]: IfcGeoScienceFeature from IfcTunnel will be covering this in another way.

File: IFC4x3/Constants/w/WATER_0SG7sVKEfA0OS6iZfD6HgM.xml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="WATER_0SG7sVKEfA0OS6iZfD6HgM" Name="WATER" UniqueId="1c407d9f-50ea-4a01-8706-b23a4d191a96">
+	<Documentation>Representation of the concept of an identified discrete water filled geological or surface feature including lakes, rivers and seas.</Documentation>
+</DocConstant>
+
```
Comment (larswik) [inline IFC4x3/Constants/w/WATER_0SG7sVKEfA0OS6iZfD6HgM.xml:5]: IfcGeoScienceFeature from IfcTunnel will be covering this in another way.

File: IFC4x3/Constants/v/VEGETATION_039IZnw4P17vEJksQmuxAR.xml (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/v/VEGETATION_039IZnw4P17vEJksQmuxAR.xml:5]: Definitely not. But this is what was submitted to ISO, so we should complain to ISO I guess.

File: IFC4x3/Constants/s/SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4.xml (hunk 3 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/s/SOIL_BORING_POINT_3R7zb_0j98qwjdYOilbWE4.xml:3]: Same as above. 😄

File: IFC4x3/Constants/s/SOLID_0hCk8da6HBZAf0CKpzqv2y.xml (hunk 4 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/s/SOLID_0hCk8da6HBZAf0CKpzqv2y.xml:5]: Agreed.

File: IFC4x3/Constants/t/TERRAIN_2TKelKxjbD$POIgGNYWWXi.xml (hunk 5 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/t/TERRAIN_2TKelKxjbD$POIgGNYWWXi.xml:3]: Agreed.

File: IFC4x3/Constants/v/VOID_2hj_ExWTf21wnmjHgzcx24.xml (hunk 6 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/v/VOID_2hj_ExWTf21wnmjHgzcx24.xml:5]: Agreed.

File: IFC4x3/Constants/w/WATER_0SG7sVKEfA0OS6iZfD6HgM.xml (hunk 7 above)
Comment (SergejMuhic) [reply IFC4x3/Constants/w/WATER_0SG7sVKEfA0OS6iZfD6HgM.xml:5]: Agreed.

**Comment by SergejMuhic** [comment]
@larswik I have resolved the conflicts. Can you resolve the open conversations? Can't recall your rights. 😃

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechnicalStratum/DocEntity.xml (hunk 1 above)
Comment (SergejMuhic) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeotechnicalStratum/DocEntity.xml:2]: I guess this was intended to be used for many things but currently it specifies IsAbstract. 0x20 determines non abstract.

**Review by larswik — APPROVED** [review]

**Comment by pjanck** [comment]
Please resolve the conflicts, then I believe we can merge.

**Pull request merged**